libtelemetry/src/main/java/com/mapbox/android/telemetry/GzipRequestInterceptor.java

```diff
@@ -13,14 +15,16 @@
 import okio.Okio;
 
 final class GzipRequestInterceptor implements Interceptor {
+  private static final int THREAD_ID = 10000;
 
   @Override
   public Response intercept(Interceptor.Chain chain) throws IOException {
     Request originalRequest = chain.request();
     if (originalRequest.body() == null || originalRequest.header("Content-Encoding") != null) {
       return chain.proceed(originalRequest);
     }
-
+    //Fix strict mode issue, see https://github.com/square/okhttp/issues/3537#issuecomment-431632176
+    TrafficStats.setThreadStatsTag(THREAD_ID);
```

Comment:
I'm a bit confused why this has to be part of the interceptor logic? Another consideration do we really have to do this for all builds? Since this is a workaround to silence `StrictMode` should we do this just for debug builds? Also should we do `TrafficStats #clearThreadStatsTag` what will happen if we don't do this?

Reply:
This must be done in the request thread, this method is in the request thread. Let's put it only in debug mode. Whether we set or clear tag will not provide additional value except for silencing Strict mode waring.

Reply:
@Chaoba let's tag only on `debug` builds and also on api levels above X.

Reply:
You mean 2.2X api level 8? If so we don't need to worry about it since the minSdkVersion of telemetry is 14 which is right the version that this method is added.